xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererShaders.cpp

```diff
@@ -132,7 +132,13 @@ void CRendererShaders::UpdateVideoFilters()
   if (!m_colorShader)
   {
     m_colorShader = std::make_unique<CYUV2RGBShader>();
-    if (!m_colorShader->Create(m_format, AVCOL_PRI_BT709, m_srcPrimaries))
+
+    AVColorPrimaries dstPrimaries = AVCOL_PRI_BT709;
+
+    if (DX::DeviceResources::Get()->Is10BitSwapchain())
+      dstPrimaries = m_srcPrimaries;
+
+    if (!m_colorShader->Create(m_format, dstPrimaries, m_srcPrimaries))
```

Comment:
10bit swapchain doesn't mean that color transformation must be disabled.

Reply:
Yes because destination supports both BT.2020 and 709 when HDR is ON

if (10bitSwapchain)
souce          /              destination
BT.2020   -----------------> BT.2020
Rec 709  ------------------> Rec 709

if  (!10bitSwapchain)
souce / dest
2020  ------------------> 709
709 --------------------> 709

In old code destination is fixed at 709 constant. This NOT works for HDR because destination needs BT.2020 = source
If 10bitSwapchain = false   nothing changes = Leia

Reply:
> if (10bitSwapchain)
> souce / destination
> BT.2020 -----------------> BT.2020
> Rec 709 ------------------> Rec 709

wrong.

Reply:
Done